server/routes/sitemap.xml.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SitemapStream, streamToPromise } from 'sitemap'
+import { joinURL } from 'ufo'
+import { serverQueryContent } from '#content/server'
+// import appConfig from '~~/app.config'
+const appConfig = useAppConfig()
+
+export default defineEventHandler(async (event) => {
+  // Fetch all documents
+  const docs = await serverQueryContent(event)
+    .where({ _partial: false, _draft: false })
+    .find()
+  const sitemap = new SitemapStream({
+    hostname: appConfig.exactproDocs.seo?.sitemap?.baseUrl ?? ''
+  })
+
+  for (const doc of docs) {
+    sitemap.write({
+      url: joinURL(
+        appConfig.exactproDocs.seo?.sitemap?.prefix ?? '',
+        String(doc._path)
+      ),
+      changefreq: 'monthly'
+    })
```

Comment:
It is better to get values and process edge cases before joining.

```suggestion
    const prefix = appConfig.exactproDocs.seo?.sitemap?.prefix ?? ''
    const path = doc._path
    if (!path) {
      continue
    }
    sitemap.write({
      url: joinURL(prefix, path),
      changefreq: 'monthly'
    })
```